### ISSUE TITLE
Bug: eager SVD sum(s) backward panics (trace grad works)

### ISSUE THREAD

**Issue by terasakisatoshi** [tensor4all/tenferro-rs#1056]
## Summary

`EagerTensor::backward()` panics for `sum(singular_values(svd(A)))` on CPU, while the traced `AdContext::grad()` path for the same computation succeeds.

This shows up in `tenferro-benchmark` `publication_gate` as `grad_sum_svd_s` / `backward` with status `panic` for tenferro-eager, while tenferro-trace is `ok`.

Reproduced at `db1990549801351308b631aef1bbca292d11a457` (2026-06-14).

## Expected

Eager reverse-mode AD should compute `d loss / d A` without panicking, matching trace `grad` (and PyTorch/JAX baselines).

## Actual

Panic during eager backward:

```
thread 'main' panicked at crates/tenferro-ad/src/eager_builder.rs:70:32:
EagerPrimitiveBuilder: missing external Derived { operation: OperationKey { operation: DotGeneral { config: DotGeneralConfig { lhs_contracting_dims: [1], rhs_contracting_dims: [0], lhs_batch_dims: [], rhs_batch_dims: [] } }, inputs: [Derived { operation: OperationKey { operation: DotGeneral { ... }, inputs: [Derived { operation: OperationKey { operation: Transpose { perm: [1, 0] }, inputs: [Derived { operation: OperationKey { operation: Extension(LinalgExtensionOp { op: Svd { eps: 0.0 } }), inputs: [Input(User { id: 3 })], role: Primary, ... }, output_slot: 0 }], ... }, output_slot: 0 }, Input(Tangent { of: User { id: 3 }, pass: 0 })], role: Linearized { active_mask: [false, true] }, ... }, output_slot: 0 }, Derived { ... Svd ... role: Linearized { active_mask: [true, false] } ... }], role: Linearized { active_mask: [true, false] }, ... }, output_slot: 0 }
```

Backtrace (abbreviated):

```
tenferro_ad::eager_builder::EagerPrimitiveBuilder::external_tensor
tenferro_ops::ad::registry::transpose_extract_diag
StdTensorOp::try_linear_transpose_rule
tidu::linear_transpose::try_linear_transpose_with_builder
TenferroBackwardCallbacks::run_transposed_linear
tidu::eager::backward::try_backward
tenferro_ad::eager::EagerTensor::backward
```

Looks like eager backward hits a linearized SVD subgraph (`Linearized { active_mask: ... }` over `LinalgExtensionOp::Svd`) but `EagerPrimitiveBuilder` has no external handler for the derived `DotGeneral` nodes emitted on that path.

## MWE (suggested regression test for `tenferro-linalg/tests/eager_tensor.rs`)

```rust
#![cfg(feature = "autodiff")]

use std::sync::{Arc, OnceLock};
use tenferro_ad::{AdContext, EagerRuntime, EagerTensor, Tensor};
use tenferro_cpu::CpuBackend;

fn ad_test_ctx() -> Arc<EagerRuntime> {
    static CTX: OnceLock<Arc<EagerRuntime>> = OnceLock::new();
    CTX.get_or_init(|| {
        let ad = AdContext::builder()
            .with_core_rules()
            .with_extension_rules(tenferro_linalg::ad_rules().unwrap())
            .build()
            .unwrap();
        EagerRuntime::with_cpu_backend_and_ad_context(CpuBackend::new(), &ad)
    })
    .clone()
}

fn well_conditioned_4x4() -> Vec<f64> {
    // Same construction as tenferro-benchmark publication_gate (seed=10).
    let n = 4;
    let mut data: Vec<f64> = (0..n * n)
        .map(|i| {
            let x = (i as u64)
                .wrapping_mul(6364136223846793005)
                .wrapping_add(10_u64.wrapping_mul(1442695040888963407));
            ((x % 1024) as f64 - 512.0) / 512.0
        })
        .collect();
    for j in 0..n {
        for i in 0..n {
            data[i + n * j] *= 0.05;
        }
        data[j + n * j] += 1.0 + j as f64 / n as f64;
    }
    data
}

#[test]
fn svd_singular_value_sum_backward_does_not_panic() {
    let ctx = ad_test_ctx();
    let a = EagerTensor::requires_grad_in(
        Tensor::from_vec_col_major(vec![4, 4], well_conditioned_4x4()),
        ctx,
    );
    let (_, s, _) = tenferro_linalg::eager_tensor::svd(&a).unwrap();
    let loss = s.reduce_sum(&[0]).unwrap();
    loss.backward().unwrap();
    assert!(a.grad().is_some());
}
```

Run:

```bash
cargo test -p tenferro-linalg --features autodiff,cpu-faer svd_singular_value_sum_backward_does_not_panic -- --nocapture
```

## Control: trace grad succeeds

```rust
use tenferro_runtime::TracedTensor;

let a = TracedTensor::from_vec_col_major(vec![4, 4], well_conditioned_4x4());
let ad = AdContext::builder()
    .with_core_rules()
    .with_extension_rules(tenferro_linalg::ad_rules().unwrap())
    .build()
    .unwrap();
let (_, s, _) = tenferro_linalg::svd(&a).unwrap();
let loss = s.reduce_sum(&[0]);
let grad = ad.grad(&loss, &a).unwrap(); // ok
```

(`tenferro-linalg/tests/traced_ad_explicit.rs` already covers SVD JVP/VJP/grad on trace; eager has no analogous backward test.)

## Notes

- Forward `svd` and batched `solve` eager backward work; only this SVD-scalar-loss path is known to panic in benchmark runs.
- Trace `jvp`/`vjp`/`grad` for the same loss are fine; gap is specific to eager tape execution / external op registration.

## External repro

`tenferro-benchmark` also ships `examples/svd_backward_panic_mwe.rs`:

```bash
RUST_BACKTRACE=1 cargo run --example svd_backward_panic_mwe --features cpu-faer
```

**Comment by terasakisatoshi** [comment]
## Repro confirmed (2026-06-14)

Still reproduces on `db1990549801351308b631aef1bbca292d11a457`.

External runnable MWE lives in [tenferro-benchmark](https://github.com/tensor4all/tenferro-benchmark):

```bash
git clone https://github.com/tensor4all/tenferro-benchmark.git
cd tenferro-benchmark
./scripts/setup_extern_deps.sh
RUST_BACKTRACE=1 cargo run --release --example svd_backward_panic_mwe --features cpu-faer
```

Observed output:

```text
trace grad sum(svd(A)) on 4x4 ...
ok
eager backward sum(svd(A)) on 4x4 ...

thread 'main' panicked at crates/tenferro-ad/src/eager_builder.rs:70:32:
EagerPrimitiveBuilder: missing external Derived { ... Extension(LinalgExtensionOp { op: Svd { eps: 0.0 } }) ... role: Linearized { active_mask: [false, true] } ... DotGeneral ... }
```

Also visible in `tenferro-benchmark` CPU ops runs as `grad_sum_svd_s_backward` / tenferro-eager = `panic`, tenferro-trace = `ok`.
